### PR TITLE
83010 - Appointment Detail null States

### PR DESCRIPTION
**Pull request by ryanshaw**
## Summary

This updates appointment detail pages to better handle situations where the appointments application does not recieve various appointment, location and facility data.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83010

## Testing done

- Manual Testing
- Updated unit tests

## Local Testing
To help with local testing and validation, new mocks have been added to [`src/applications/vaos/services/mocks/index.js`](https://github.com/department-of-veterans-affairs/vets-website/blob/eef9af7cf8684e6afe15cb77c049501786ebf15e/src/applications/vaos/services/mocks/index.js#L35)

Here's the [Figma file](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2159-51275&t=pxGzhSukzuC02chF-1)

## Screenshots

|              | With Data | Missing Data |
| ------- | ------- | ----- |
| In-Person |image|image|
|Video|image|image|
|VA Video|image|image|
|ATLAS|image|image|
|Community Care|image|image|
|Phone|image|image|
|Claim Exam|image|image|

## What areas of the site does it impact?

This impacts the following appointment detail pages:
- In-person appointment
- Video
- Video at VA
- ATLAS
- Phone
- Community Care
- Claim Exam

## Acceptance criteria

Background: the `va_online_scheduling_appointment_details_redesign` toggle is on

- [ ] Given when the services return a null value for type of care for all appointment types
When the user is on the Appointment Details page
Then the label `What` will not be displayed 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

- [ ] Given when the services return a null value for the facility (but receive the facility ID) for all appointment types
When the user is on the Appointment Details page
Then the frontend will display the facility name
And a link to facility locator 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

- [ ] Given when the services return a null value for the facility (no facility ID) for all appointment types
When the user is on the Appointment Details page
Then the frontend will display the text `Facility details not available`
And a link to the facility locator 
And page design matches the [design spec](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=2072-398573&t=FCm7ZccXpWJX1nu8-0)

### Quality Assurance & Testing

- [ ] I |updated and added unit tests and integration tests for each feature 
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

**Review by jenniemc — COMMENTED** [review]
For video atlas appt, when there is missing address location it currently displays "Where to attend" heading but there's no text underneath. Should there be message that say something like "address is not available"? Everything else is looking good.
Screenshot 2024-07-09 at 4 01 50 PM

**Comment by JunTaoLuo** [comment]
A few thoughts/questions after going through the figma and the screenshots above:

## General

In several of appointment types, there's a section for "Who" that should display the provider name when the data is available. I don't see those sections in the "With Data" screenshots. Is that intentional?

## VA Video appointments

For VA Video appointments, I think we should check with the designers what's the expectation for the "Need to make changes?" section. Specifically, like the Video appointments, the specification at the top of the figma file says:
image
Which indicates that "Facility not available" and "Clinic not available" should be displayed but, unlike the Video appointments, the example at the bottom of the figma file does not have this section. 
image

## ATLAS

In the specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

## Phone appointments

For phone appointments, the figma files says:
image
Is that something we need to add?
In the specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

## Community Care

This is a question for the designers. For every other appointment type that has provider information, it is displayed under the "Who" heading but Community Care appointments display it under the "Provider" heading. Is this intentional?

## Claim Exam

The screenshot still has a "What" section can you take a screenshot of when that's missing? Also, the specifications calls for using the main VA number when the facility phone is unavailable.

I'll start digging into the code and business logic next.

**Comment by jenniemc** [comment]
Good catch @JunTaoLuo 

> In several of appointment types, there's a section for "Who" that should display the provider name when the data is available. I don't see those sections in the "With Data" screenshots. Is that intentional?

Some appointment types such as in-person and claims exam, did not bake the "who" into the layout. We need to create a new ticket to ensure those appointments display the Who.

> In the ATLAS specification, there's a section for "What" but I didn't see it in the With Data screenshot. Is that intentional?

The set of mock data are pull directly from staging. If the `serviceType` (what) for ATLAS is missing it is because it was not returned. We need to check the vets-api If ATLAS appt has the serviceType property returned. If so we can simply add serviceType into our mock ATLAS data.

> The Claim exam screenshot still has a "What" section can you take a screenshot of when that's missing? 

For Claim exam appointments, the "What" will always default to Claim exam and will ignore the `serviceType` property. Therefore the "What" will always be present 

> Also, the specifications calls for using the main VA number when the facility phone is unavailable.

Phone numbers will be fixed in the ticket, [#80375](https://github.com/department-of-veterans-affairs/va.gov-team/issues/80375)

**Comment by JunTaoLuo** [comment]
> For Claim exam appointments, the "What" will always default to Claim exam and will ignore the serviceType property. Therefore the "What" will always be present

Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?

> The set of mock data are pull directly from staging. If the serviceType (what) for ATLAS is missing it is because it was not returned. We need to check the vets-api If ATLAS appt has the serviceType property returned. If so we can simply add serviceType into our mock ATLAS data.

Ah I see, let me see if I can dig into this a bit in vets-api. Might take me a minute to figure it out though.

**Comment by jenniemc** [comment]
> Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?

When I saw the [figma mockup](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=3270-47171&t=h5AfSLX3paE54SUd-0) I thought it was repetitive as well to have Claim exam repeated in the title and in the "What". This would be a great question to ask @Cieramaddox or @outerpress if it is intentional.

**Comment by ryanshaw** [comment]
> For video atlas appt, when there is missing address location it currently displays "Where to attend" heading but there's no text underneath. Should there be message that say something like "address is not available"? Everything else is looking good. Screenshot 2024-07-09 at 4 01 50 PM

@jenniemc Just pushed a fix for this and updated screenshots

**Comment by ryanshaw** [comment]
> > Hmm does this make sense though? Maybe we should check with the designers but it seems redundant since the title already says Claim exam. Seems to me that repeating it in the what section doesn't add any value?
> 
> When I saw the [figma mockup](https://www.figma.com/design/q1lq0bwvDMCg6RADZddYMX/Details-Pages-Specs-(WIP)-%7C-Appointments-FE?node-id=3270-47171&t=h5AfSLX3paE54SUd-0) I thought it was repetitive as well to have Claim exam repeated in the title and in the "What". This would be a great question to ask @Cieramaddox or @outerpress if it is intentional.

@jenniemc @JunTaoLuo I talked to @Cieramaddox about this and it is intentional to have both for consistency purposes. We can revisit it at a later date if needed.

**Comment by jenniemc** [comment]
ATLAS video appointments are sometimes held at a non VA medical facility such as an American Legion Post or at a Veteran Foreign War Post. The "find facility link" takes the user to VA Service type and not necessarily to ATLAS locator. A better link would be https://telehealth.va.gov/atlas. Please confirm with the Ciera which link to use and what language to use for missing ATLAS location address.

File: src/applications/vaos/components/layout/InPersonLayout.jsx [hunk 1]
```diff
@@ -64,24 +68,43 @@ export default function InPersonLayout({ data: appointment }) {
...
+              <span>Facility details not available</span>
+              <br />
+              <NewTabAnchor href="/find-locations">
+                Find facility information
+              </NewTabAnchor>
+              <br />
+              <br />
+            </>
+          )}
+        {/* When the services return a null value for the facility (but receive the facility ID) */}
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/layout/InPersonLayout.jsx:91]: In the spec it says:

image

Is this the facility name something we can retrieve using just the ID?

File: src/applications/vaos/components/layout/InPersonLayout.jsx [hunk 2]
```diff
@@ -98,7 +121,7 @@ export default function InPersonLayout({ data: appointment }) {
         <span>Location: {clinicPhysicalLocation || 'Not available'}</span>{' '}
         <br />
         {facilityPhone && (
-          <FacilityPhone heading="Clinic phone:" contact={facilityPhone} />
+          <FacilityPhone heading="Phone:" contact={facilityPhone} />
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/layout/InPersonLayout.jsx:124]: Hmm, the spec seems to prefer clinic phone and falls back to facility phone if that's missing. Do we need to update the logic here?

image

**Review by JunTaoLuo — COMMENTED** [review]
Looks great so far. Just had a few more questions in the code. Looking at the tests now and that should be the last bit of review I need to do.

File: src/applications/vaos/components/layout/CCLayout.jsx [hunk 3]
```diff
@@ -57,12 +57,14 @@ export default function CCLayout({ data: appointment }) {
               </div>
             )}
         </When>
-        <What>{typeOfCareName || 'Type of care not noted'}</What>
+        <What>{typeOfCareName}</What>
         <Section heading="Provider">
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/layout/CCLayout.jsx:61]: Hey @Cieramaddox I see that in the figma that all other appointments uses the "Who" heading to display the provider information with CC being the only type that uses the "Provider" heading. Just wanted to double check if this is intentional.

File: src/applications/vaos/components/tests/ClaimExamLayout.unit.spec.js [hunk 4]
```diff
@@ -33,6 +33,66 @@ describe('VAOS Component: ClaimExamLayout', () => {
     },
   };
 
+  describe('When appointment information is missing', () => {
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/tests/ClaimExamLayout.unit.spec.js:36]: Can we add another case for when facility ID is present but facility info is missing? This goes for all other appointment types that has a different "View facility information"/"Find facility information" distinction.

File: src/applications/vaos/components/layout/InPersonLayout.jsx (hunk 2 above)
Comment (ryanshaw) [reply src/applications/vaos/components/layout/InPersonLayout.jsx:124]: @JunTaoLuo All the phone number issues are in another issue that @jenniemc is working on (https://github.com/department-of-veterans-affairs/va.gov-team/issues/80375)

File: src/applications/vaos/components/tests/PhoneLayout.unit.spec.js [hunk 5]
```diff
@@ -33,6 +33,78 @@ describe('VAOS Component: PhoneLayout', () => {
...
+          name: /Who/i,
+        }),
+      ).not.to.exist;
+
+      expect(
+        screen.queryByRole('heading', {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/tests/PhoneLayout.unit.spec.js:104]: Also need to add assertions for the text for missing facility/clinic info as well as the reason/other details texts.

File: src/applications/vaos/components/tests/VideoLayoutAtlas.unit.spec.js [hunk 6]
```diff
@@ -33,6 +33,79 @@ describe('VAOS Component: VideoLayoutAtlas', () => {
...
+          name: /Who/i,
+        }),
+      ).not.to.exist;
+
+      expect(
+        screen.queryByRole('heading', {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/tests/VideoLayoutAtlas.unit.spec.js:105]: Add assertions for missing facility and clinic info.

File: src/applications/vaos/components/tests/VideoLayoutVA.unit.spec.js [hunk 7]
```diff
@@ -33,6 +33,68 @@ describe('VAOS Component: VideoLayoutVA', () => {
...
+          name: /Who/i,
+        }),
+      ).not.to.exist;
+
+      expect(
+        screen.queryByRole('heading', {
+          level: 2,
+          name: /What/i,
+        }),
+      ).not.to.exist;
```
Comment (JunTaoLuo) [inline src/applications/vaos/components/tests/VideoLayoutVA.unit.spec.js:95]: Need to add assertions for missing facility and clinic info. Also consider adding another case where facility id is present but missing facility info.

**Review by JunTaoLuo — COMMENTED** [review]
Looks good. Some additional testing details to add.

File: src/applications/vaos/components/layout/InPersonLayout.jsx (hunk 2 above)
Comment (JunTaoLuo) [reply src/applications/vaos/components/layout/InPersonLayout.jsx:124]: Ah I see it's tracked in the same issue. Thanks!

File: src/applications/vaos/components/layout/ClaimExamLayout.jsx [hunk 8]
```diff
@@ -69,25 +72,44 @@ export default function ClaimExamLayout({ data: appointment }) {
...
-            <span>Facility details not available</span>
-            <br />
-            <NewTabAnchor href="/find-locations">
-              Find facility information
-            </NewTabAnchor>
-            <br />
-            <br />
-          </>
-        )}
+        {/* When the services return a null value for the facility (no facility ID) for all appointment types */}
```
Comment (ryanshaw) [inline src/applications/vaos/components/layout/ClaimExamLayout.jsx:83]: The facility logic in `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx` should probably be pulled out and placed in a component on its own. Could simplify unit tests as well as future code changes. Didn't want to introduce that refactor this late in the review process so will open another issue for that.

File: src/applications/vaos/components/layout/InPersonLayout.jsx (hunk 1 above)
Comment (ryanshaw) [reply src/applications/vaos/components/layout/InPersonLayout.jsx:91]: Current behavior (on old design) doesn't display the name and I cannot find anywhere that would exist if we only have the id. My assumption is that we wouldn't be getting that information back from the facilities endpoint but would be able to construct the link with the id to allow users to get more information about that facility.

File: src/applications/vaos/components/tests/ClaimExamLayout.unit.spec.js (hunk 4 above)
Comment (ryanshaw) [reply src/applications/vaos/components/tests/ClaimExamLayout.unit.spec.js:36]: @JunTaoLuo I've added tests cases for this for `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx`

File: src/applications/vaos/components/tests/PhoneLayout.unit.spec.js (hunk 5 above)
Comment (ryanshaw) [reply src/applications/vaos/components/tests/PhoneLayout.unit.spec.js:101]: @JunTaoLuo Added assertions for the phone appointments

File: src/applications/vaos/components/tests/VideoLayoutAtlas.unit.spec.js (hunk 6 above)
Comment (ryanshaw) [reply src/applications/vaos/components/tests/VideoLayoutAtlas.unit.spec.js:110]: @JunTaoLuo I've added these assertions

File: src/applications/vaos/components/tests/VideoLayoutVA.unit.spec.js (hunk 7 above)
Comment (ryanshaw) [reply src/applications/vaos/components/tests/VideoLayoutVA.unit.spec.js:207]: @JunTaoLuo I've added tests cases for this (the facility id/missing facility scenario) for `ClaimExamLayout.jsx`, `VideoLayoutVA.jsx` and `InPersonLayout.jsx`

File: src/applications/vaos/components/layout/InPersonLayout.jsx (hunk 1 above)
Comment (JunTaoLuo) [reply src/applications/vaos/components/layout/InPersonLayout.jsx:91]: That's fair enough, but might want to check with @Cieramaddox to make sure.

**Review by JunTaoLuo — APPROVED** [review]
LGTM! There just two things to check with design but otherwise I think this is ready 😄. Hey @vbahinwillit @jenniemc can you take a look at this as well?

**Review by jenniemc — APPROVED** [review]
LGTM! Let it merge!!

**Pull request merged**